django-backend/fecfiler/transactions/models.py

```diff
@@ -179,6 +182,96 @@ def save(self, *args, **kwargs):
 
         super(Transaction, self).save(*args, **kwargs)
 
+    def delete(self):
+        if self.can_delete and not self.deleted:
+            self.deleted = datetime.now()
```

Comment:
because the model inherits SoftDeleteModel, you can replace 186,187 with `super().delete()`
Then all the other instances of 
```
x.deleted = datetime.now()
x.save()
```
can be replaced with `x.delete()`